app/controllers/registrations/school_profile_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class Registrations::SchoolProfileController < ApplicationController
+  def show
+    @school_profile_form = SchoolProfileForm.new
+  end
+
+  def create
+    @school_profile_form = SchoolProfileForm.new(school_params)
+
+    if @school_profile_form.valid?
+      load_school
+      redirect
+    else
+      render :show
+    end
+  end
+
+private
+
+  def school_params
+    params.require(:school).permit(:urn)
+  end
+
+  def load_school
+    session["school_urn"] = school_params[:urn]
+    @school = School.find_by(urn: school_params[:urn])
+  end
+
+  def redirect
+    if !@school.eligible?
```

Comment:
could we create a helper method for not_eligable?

Reply:
In here it is hust a rails method. But that will be changing with some of the work Ethan is doing. I don't think a helper would be correct for domain knowledge too.

Reply:
What about `unless`?

Reply:
Bad pattern @ethanmills mixing if and unless - it takes more mental computation when reading the code you either stick to one or another

Reply:
Bad pattern is Ruby including unless imo.

But while it's there, a one line method using `unless` while some other methods use `if` doesn't look like a bad pattern to me.

Reply:
For me personally, I find it easier to read an `unless` than an `if !`. But if we don't ever want to use unless, then we should enforce that through linting.

Reply:
Ruby had a lofty go to read just as natural language hence rubysims like unless. That said, I find it ok as a guard in a method call but would steer well clear of unless xxxx else xx end or as in this case, using it with a load of other conditionals. And it could just be my bad English but sometimes I kind of forget what unless means.

Reply:
Yep, I agree, I hadn't realised this was the start of a chain of elsifs